plugin/clap.vim

```diff
@@ -19,3 +19,11 @@ augroup ClapBuffers
   autocmd BufWinEnter,WinEnter * let g:__clap_buffers[bufnr('')] = reltimefloat(reltime())
   autocmd BufDelete * silent! call remove(g:__clap_buffers, expand('<abuf>'))
 augroup END
+
+" yanks provider
+if get(g:, 'clap_enable_yanks_provider', 1)
+  augroup ClapYanks
+    autocmd!
+    autocmd VimEnter * call clap#provider#yanks#init()
+  augroup END
```

Comment:
We could avoid sourcing the yanks provider on VimEnter by introducing a global variable, although the sourcing is actually cheap :(.

```vim
if get(g:, 'clap_enable_yanks_provider', 1)
  let g:__clap_last_yanked = getreg('"')
  augroup ClapYanks
    autocmd!
    autocmd TextYankPost * call clap#provider#yanks#collect()
  augroup END
endif
```

Initialize `s:yank_history` with `g:__clap_last_yanked`:


```vim
let s:yank_history = exists('g:__clap_last_yanked') ? [g:__clap_last_yanked] : []
```

Expose `s:yanks_collect()` to `clap#provider#yanks#collect()`, and then we can remove `clap#provider#yanks#init()`.

Reply:
`getreg()` in the above code executes before reading of `viminfo/shada` file before the VimEnter event.  As a result, yank provider starts with an empty list(even if register `"` is nonempty). Hence we may need to live with VimEnter autocmd for now. 

I have exposed `s:yanks_collect()` to `clap#provider#yanks#collect()`
I have implemented a max. yank entries through global variable(Default:20)